docs/design/auditing-containment.md

```diff
@@ -0,0 +1,168 @@
+# Auditing and Node Containment
+
+## Abstract
+
+This design doc outlines how we will implement "containment mode," in which the auditing service monitors a node that has accepted an audit request, but refuses to send the requested data.
+Currently in the codebase, nodes that do this will be marked as having audit failures.
+However, we don't want to mark nodes as offline or immediately mark an audit failure if this specific case arises.
+Instead, we want to "contain" these nodes by retrying the original audit until they either eventually respond and pass the audit, or refuse to respond a certain number of times and ultimately receive an audit failure mark.
+
+## Main Objectives
+1. Nodes won’t be able to “escape” audits for specific data by refusing to send requested data.
+2. The audit service should not audit contained nodes for other data, and should verify their originally requested data.
+3. The overlay service should not select contained nodes.
```

Comment:
What's the exact motivation for not selecting contained nodes? It seems to be worse to be in containment mode than failing audit requests.

Reply:
The problem comes with cheaters like me. In the past I was able to avoid audit penalites by dropping the connection without sending data or keep the connection open without sending data. The audit job was just hanging. Mission completed.

In order to stop cheaters like me you have to give me audit penalties. That should be our current implementation. The downside of that is that it will hit honest storage nodes as well. The containment mode will give these nodes a second and maybe even more chances while my cheater node is out of the game.

Reply:
But this is the reason to introduce the containment mode, right? After a few attempts to re-request the audited erasure share, your node will fail the audit and it will get the penalty.

I am also not sure how long it will take until a contained node will be re-checked in a real-life system. 30 minutes, 1 day, 1 year?

Reply:
We have to hit each storage nodes at least a few times per day.